core/src/qos_service.rs

```diff
@@ -153,10 +153,15 @@ impl QosService {
                             self.metrics.retried_txs_per_account_limit_count.fetch_add(1, Ordering::Relaxed);
                             Err(TransactionError::WouldExceedMaxAccountCostLimit)
                         }
-                        CostTrackerError::WouldExceedAccountDataMaxLimit => {
+                        CostTrackerError::WouldExceedAccountDataBlockLimit => {
                             self.metrics.retried_txs_per_account_data_limit_count.fetch_add(1, Ordering::Relaxed);
                             Err(TransactionError::WouldExceedMaxAccountDataCostLimit)
                         }
+                        CostTrackerError::WouldExceedAccountDataTotalLimit => {
+                            // NOTE: Do *not* increment a retry counter, since these
+                            // transactions will *not* be retried
```

Comment:
Yes, I can do that.

Reply:
We probably don't want a datapoint on every occurrence of this, seeing as how every transaction would cause an instance of this datapoint to be reported. 

Maybe just the first time, then set a flag to not report it again?

Reply:
> We probably don't want a datapoint on every occurrence of this, seeing as how every transaction would cause an instance of this datapoint to be reported when we hit the limit.

Yeah, this is why I didn't have a datapoint originally.

> One way to approach this is to aggregate these errors in to the QosServiceMetrics in accumulate_estimated_transaction_costs() where the cost model results are being used
> and report `qos-service-errors`, a parallel to the existing `qos-service-stats`

Right now I've added `dropped_txs_per_account_data_total_limit_count` to `qos-service-stats`. Is that sufficient, or would you like me to move this datapoint to a `qos-service-errors`?

Reply:
I think a separate datapoint makes sense because errors are a good logical separation. I would even go as far as to say it can belong to a separate struct within the existing metrics, kind of like this:https://github.com/solana-labs/solana/blob/619335df1a87b58b7c7d07de20bcb3f56b58b925/core/src/leader_slot_banking_stage_metrics.rs#L218-L233 is broken up into different structs.

 Otherwise these datapoints start to grow to a size where they become burdensome to manage

Reply:
Ok, done! One nice result from moving the metrics accumulation into
https://github.com/solana-labs/solana/blob/35d1235ed0d22b0f46f5cd3be543978c2a7569e0/core/src/banking_stage.rs#L1340-L1345
is that atomics are no longer updated _per transaction_, and now its _per batch_.